### PR TITLE
[EAPQE-2697] Add test for MicroProfile LRA with Fault Tolerance

### PR DESCRIPTION
**Pull request by mnovak1**
Please make sure your PR meets the following requirements:
- [x] Pull Request contains a description of the changes
- [x] Pull Request does not include fixes for multiple issues/topics
- [x] Code is formatted, imports ordered, code compiles and tests are passing
- [x] Link to the passing job is provided
- [x] Code is self-descriptive and/or documented

CI run: https://url.corp.redhat.com/8fbe2ad

### CODE REVIEW

File: microprofile-lra/src/main/java/org/jboss/eap/qe/microprofile/lra/LRAParticipantWithFaultTolerance1.java [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+        lraResult.setLraId(lraId);
+        lraResult.setRecoveryId(recoveryId);
+
+        // call Participant 2 to propagate the LRA
+        try (Client client = ClientBuilder.newClient()) {
+            client.target(uriInfo.getBaseUri() + "/" + LRAParticipantWithFaultTolerance2.PATH)
+                    .request().get();
+        }
+        if (failLRA) {
+            throw new Exception("Throwing exception to fail participant until circuit is open.");
```
Comment (fabiobrz) [inline microprofile-lra/src/main/java/org/jboss/eap/qe/microprofile/lra/LRAParticipantWithFaultTolerance1.java:160]: How is circuit breaker related here? Shouldn't the message be about FT Fallback?

File: microprofile-lra/src/test/java/org/jboss/eap/qe/microprofile/lra/FaultToleranceWithLRATest.java [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(Arquillian.class)
+@ServerSetup({ EnableLRASetupTask.class, FaultToleranceServerSetup.class })
+public class FaultToleranceWithLRATest {
```
Comment (fabiobrz) [inline microprofile-lra/src/test/java/org/jboss/eap/qe/microprofile/lra/FaultToleranceWithLRATest.java:34]: Can some minimal Javadoc be dded for this class and its test methods?

**Review by fabiobrz — CHANGES_REQUESTED** [review]
Thanks for this PR @mnovak1 
Changes make sense to me but Ive dropped some comments, let me know what you think.

BTW -  I've been working on some common tooling for MP Telemetry and Micrometer as well, related to server config, and I put classes in tooling-observability, but I think I'll move those to tooling-server-config, as you did, which seems the actual common place for modules to use one another's server config tooling.

**Review by mnovak1 — COMMENTED** [review]
Fixed and repushed.

File: microprofile-lra/src/main/java/org/jboss/eap/qe/microprofile/lra/LRAParticipantWithFaultTolerance1.java [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+                    .entity("Wrong recoveryId across retries. Expected " + lraResult.getRecoveryId() + " but got " + recoveryId)
+                    .build();
+        }
+
+        lraResult.setLraId(lraId);
+        lraResult.setRecoveryId(recoveryId);
+
+        // call Participant 2 to propagate the LRA
+        try (Client client = ClientBuilder.newClient()) {
+            client.target(uriInfo.getBaseUri() + "/" + LRAParticipantWithFaultTolerance2.PATH)
```
[the remark this answers is not in the document]
Comment (fabiobrz) [reply microprofile-lra/src/main/java/org/jboss/eap/qe/microprofile/lra/LRAParticipantWithFaultTolerance1.java:78]: Ok, thanks

**Comment by fabiobrz** [comment]
> Fixed and repushed.

Thanks @mnovak1 - changes LGTM. I started a new job run to double check everything is still fine, despite the minimal changes.

That being said, JFYI, there are upstream (GitHub based) manual workflows too, that you can start against a WildFly fork/branch. In order to use it for testing a PR, just browse the _Actions_ at your fork, and you should be able to start either a regular or bootable JAR execution manually from there.

**Review by fabiobrz — APPROVED** [review]

**Comment by fabiobrz** [comment]
Changes LGTM, and failures in CI runs are unrelated. Merging, thanks @mnovak1!

**Pull request merged**